coffee/base.py

```diff
@@ -221,20 +217,61 @@ class ArrayInit(Expr):
     to some values. For example ::
 
         A[3][3] = {{0.0}} or A[3] = {1, 1, 1}.
+    """
+
+    _default_precision = 12
 
-    At the moment, initial values like ``{{0.0}}`` and ``{1, 1, 1}`` are passed
-    in as simple strings."""
+    def __init__(self, values, precision=None):
+        """Initialize an ArrayInit object.
 
-    def __init__(self, values):
+        :arg values: representation of the values the array is initialized to
+        :type values: a string or a numpy ndarray.
+        :arg precision: the number of decimal digits that should be used when
+            converting a float (in a numpy array) to a string.
+        :type precision: integer (defaults to 12)
+        """
         self.values = values
+        self.precision = precision or ArrayInit._default_precision
 
-    def reconstruct(self, values, **kwargs):
-        return type(self)(values, **kwargs)
+    def reconstruct(self, values, precision, **kwargs):
+        return type(self)(values, precision, **kwargs)
 
     def operands(self):
         return [self.values], {}
```

Comment:
You need to return the precision as well, otherwise it's lost when reconstructing via operands.